modules/nw-ovn-kubernetes-live-migration.adoc

```diff
@@ -0,0 +1,71 @@
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc
+// * networking/openshift_sdn/rollback-to-ovn-kubernetes.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ovn-kubernetes-live-migration_{context}"]
+= Migrating to the OVN-Kubernetes network plugin by using the live migration method
+
+The following procedure entails checking for egress router resources and using the live migration method to migrate from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin.
+
+.Prerequisites
+
+* A cluster has been configured with the OpenShift SDN CNI network plugin in the network policy isolation mode.
+* You have installed the OpenShift CLI (`oc`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have created a recent backup of the etcd database is available.
+* The cluster is in a known good state without any errors.
+* On all cloud platforms after updating software, a security group rule must be in place to allow UDP packets on port `6081` for all nodes.
+* Cluster administrators must remove any egress router resources.
+* You have reviewed the "Considerations for live migration to the OVN-Kubernetes network plugin" section of this document.
+
+.Procedure 
+
+. On the {product-title} web console, click *Observe* -> *Metrics*.
+
+. In the *Expression* box, add the following queries, then click *Add query*:
++
+[source,text]
+----
+network_attachment_definition_instances{networks="egress-router"} > 0
+----
+
+. Click *Run queries*. If any data points are found, you should not proceed with the OpenShift SDN to OVN-Kubernetes live migration procedure. These custom resources must be removed before using the live migration procedure.
+
+. To remove an egress router, you can delete the egress router custom resource definition:
++
+[source,terminal]
+----
+$ oc delete crd egressrouters.network.operator.openshift.io
+----
+
+ifdef::openshift-rosa,openshift-dedicated[]
+. Enter the following command to add the `unsupported-red-hat-internal-testing` annotation to the cluster-level network configuration:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"unsupported-red-hat-internal-testing": "true"}}}'
+----
+endif::[]
+
+. Enter the following command to patch the cluster-level networking configuration and initiate the migration from OpenShift SDN to OVN-Kubernetes:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'
+----
++
+After running these commands, the migration process begins. The time it takes to complete the process will vary depending on how many nodes are in your cluster. 
+
+. Optional: You can enter the following commands to ensure that the migration process has completed, and to check the status of the `network.config`:
```

Comment:
We added new metrics for users to monitor the migration progress in 4.16. https://docs.google.com/document/d/1QQJDtipDqSyLnOYkoEJ12QnV-g61d6BwDLY-aOpdZVw/edit

Reply:
above link show check the metrics from console. 

and we can also provide another way to check from oc client:


```
# token=`oc create token prometheus-k8s -n openshift-monitoring`
# oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" "https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?" --data-urlencode "query=openshift_network_operator_live_migration_condition" | jq`
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": [
      {
        "metric": {
          "__name__": "openshift_network_operator_live_migration_condition",
          "container": "network-operator",
          "endpoint": "metrics",
          "instance": "10.0.83.62:9104",
          "job": "metrics",
          "namespace": "openshift-network-operator",
          "pod": "network-operator-6c87754bc6-c8qld",
          "prometheus": "openshift-monitoring/k8s",
          "service": "metrics",
          "type": "NetworkTypeMigrationInProgress"
        },
        "value": [
          1717653579.587,
          "1"
        ]
      },
      {
        "metric": {
          "__name__": "openshift_network_operator_live_migration_condition",
          "container": "network-operator",
          "endpoint": "metrics",
          "instance": "10.0.83.62:9104",
          "job": "metrics",
          "namespace": "openshift-network-operator",
          "pod": "network-operator-6c87754bc6-c8qld",
          "prometheus": "openshift-monitoring/k8s",
          "service": "metrics",
          "type": "NetworkTypeMigrationMTUReady"
        },
        "value": [
          1717653579.587,
          "0"
        ]
      },
      {
        "metric": {
          "__name__": "openshift_network_operator_live_migration_condition",
          "container": "network-operator",
          "endpoint": "metrics",
          "instance": "10.0.83.62:9104",
          "job": "metrics",
          "namespace": "openshift-network-operator",
          "pod": "network-operator-6c87754bc6-c8qld",
          "prometheus": "openshift-monitoring/k8s",
          "service": "metrics",
          "type": "NetworkTypeMigrationOriginalCNIPurged"
        },
        "value": [
          1717653579.587,
          "0"
        ]
      },
      {
        "metric": {
          "__name__": "openshift_network_operator_live_migration_condition",
          "container": "network-operator",
          "endpoint": "metrics",
          "instance": "10.0.83.62:9104",
          "job": "metrics",
          "namespace": "openshift-network-operator",
          "pod": "network-operator-6c87754bc6-c8qld",
          "prometheus": "openshift-monitoring/k8s",
          "service": "metrics",
          "type": "NetworkTypeMigrationTargetCNIAvailable"
        },
        "value": [
          1717653579.587,
          "0"
        ]
      },
      {
        "metric": {
          "__name__": "openshift_network_operator_live_migration_condition",
          "container": "network-operator",
          "endpoint": "metrics",
          "instance": "10.0.83.62:9104",
          "job": "metrics",
          "namespace": "openshift-network-operator",
          "pod": "network-operator-6c87754bc6-c8qld",
          "prometheus": "openshift-monitoring/k8s",
          "service": "metrics",
          "type": "NetworkTypeMigrationTargetCNIInUse"
        },
        "value": [
          1717653579.587,
          "0"
        ]
      }
    ],
    "analysis": {}
  }
}
```